core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java

```diff
@@ -239,20 +245,29 @@ public Boolean call(BlockWorkerClientService.Client client) throws TException {
   }
 
   @Override
-  public void sessionHeartbeat() throws IOException {
-    retryRPC(new RpcCallable<Void, BlockWorkerClientService.Client>() {
-      @Override
-      public Void call(BlockWorkerClientService.Client client) throws TException {
-          client.sessionHeartbeat(mSessionId, null);
-          return null;
-      }
-    });
+  public void sessionHeartbeat() throws IOException, InterruptedException {
+    BlockWorkerClientService.Client client =
```

Comment:
better javadoc this reason why sessionHeartbeat is treated differently from other operations.